core/src/test/kotlin/fr/sncf/osrd/pathfinding/PathfindingTest.kt

```diff
@@ -623,3 +626,265 @@ class PathfindingTest : ApiTest() {
+        )
+    }
+
+    @Test
+    fun reversedNonStopsAtEndOfBlockTest() {
+        // Stops are not moved - nothing to do here
+
+        val parsed =
+            callPathfindingEndpoint(
+                TestTrains.REALISTIC_FAST_TRAIN,
+                waypoints.reversed(),
+                "small_infra/infra.json",
+                false,
+            )
+        checkPathfindingSuccess(
+            parsed,
+            45900.meters,
+            expectedIntermediatePathItemPosition =
+                listOf(
+                    Offset(reversedFirstIntermediateStopDistance),
+                    Offset(reversedSecondIntermediateStopDistance),
+                ),
+        )
+    }
+
+    @Test
+    fun shortTrainTest() {
+        // Only intermediate stops are moved to the end of their block
+        val recalculatedFirstIntermediateStopDistance =
+            firstIntermediateStopDistance + TestTrains.VERY_SHORT_FAST_TRAIN.length.meters
+        val recalculatedSecondIntermediateStopDistance =
+            secondIntermediateStopDistance + TestTrains.VERY_SHORT_FAST_TRAIN.length.meters
+
+        val parsed =
+            callPathfindingEndpoint(
+                TestTrains.VERY_SHORT_FAST_TRAIN,
+                waypoints,
+                "small_infra/infra.json",
+                true,
+            )
+        checkPathfindingSuccess(
+            parsed,
+            45900.meters,
+            expectedIntermediatePathItemPosition =
+                listOf(
+                    Offset(recalculatedFirstIntermediateStopDistance),
+                    Offset(recalculatedSecondIntermediateStopDistance),
+                ),
+        )
+    }
+
+    @Test
+    fun reversedShortTrainTest() {
+        // Only intermediate stops are moved to the end of their block
+        val recalculatedFirstIntermediateStopDistance =
+            reversedFirstIntermediateStopDistance + TestTrains.VERY_SHORT_FAST_TRAIN.length.meters
+        val recalculatedSecondIntermediateStopDistance =
+            reversedSecondIntermediateStopDistance + TestTrains.VERY_SHORT_FAST_TRAIN.length.meters
+
+        val parsed =
+            callPathfindingEndpoint(
+                TestTrains.VERY_SHORT_FAST_TRAIN,
+                waypoints.reversed(),
+                "small_infra/infra.json",
+                true,
+            )
+        checkPathfindingSuccess(
+            parsed,
+            45900.meters,
+            expectedIntermediatePathItemPosition =
+                listOf(
+                    Offset(recalculatedFirstIntermediateStopDistance),
+                    Offset(recalculatedSecondIntermediateStopDistance),
+                ),
+        )
+    }
+
+    @Test
+    fun longTrainTest() {
+        // The first intermediate stop is moved to the next block-delimiting signal, but the
+        // distance available is less than the length of the train, so it is moved to the end of the
+        // block
+        // (270 meters after the first intermediate stop)
```

Comment:
nit: you got bitten by spotless (ditto on all the comments of the file) 😿
And my other suggestions might be affected too, of course 😓
```suggestion
        // block (270 meters after the first intermediate stop)
```